### PR TITLE
Fix i3dm quantization parsing

### PR DESCRIPTION
**Pull request by ptrgags**
I've been working on converting the 3D Tiles BIM model to not use the deprecated `KHR_techniques_webgl`, but along the way I came across a bug where the `i3dm` models were not appearing in ModelExperimental (but working fine for Model).

[This Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=tVl7d9o4Fv8qmpw9J6ZDjV+8StKzDWFaspCkCU0nbeZkjC1AwdisJZOQnnz3vZIsPwhk0p3dv5Cl+/xd3asr0cWUJAu997DEMVngkLnBb9hlSYypjkN3HOBh5OOguI4OEYsT3LkJb8KuZO9Hoe7jiZsE7IPnYUpH0RyHQHizh9cns/FHj5yRk/6Xx755Svq0H17UvW6/0Z8vf7/qnrR1IPq3/3EORH3j2+jqbnjXe7heG/fXX6+Cwehicfo4Z2fHQ+eUGI+njxfB8M6/+3bcf/i2GJJB92T5DYQNR9P6cDQH4ad3Y/so8Nb9xldQ7tl9ThNer098Tnv9+2dydtczTx8/W2fHU3P4CAa4+NponLD7+eqa1IfroHfrnfitzyvcf1jdfemP3/acT+zjSWM6nt7sCc9rNXQe3eMYnQduyJAPQExDtOBgoWUcrYiPfTReoyPALMBrdLmmDC/oTehFIWVoRTBnPkQhvkcpildiTrvZ88R3NwK0SYjjm71KR/FRD4cY2CS/Lj6FPekEfmA49LVUopyUH/bxiAQY4KFL7LEoHpIHEuaCiZrnVoiY51q8rQKEdEFZNEBYpHvuAseuTjHj4rQfNyHiEDESuoxE4bui2103ZjByQ1szLdsx623dMBoNq9lsVtHbutFqGE5dt6y6YTTrrVYV2Q3btppt3Wo67abjNMxKlcuv1V6jwbbajmnpTaPZMhuNllUXOizHNBzdMOu2bZqOzXW02g3T1J1Wy2g7jZYtdUQx4TnwTMMn7PoknJ4T5s0uoiDQDL1utG0HbAfRFjjD9Ri6bbebzbrptO16y2oajSpq6FbLNlt1C/TYDcdscUVPFQGpjAzjqGMG4ZDYLiERCSMryFDX9zVuVtHXYqwwk9BzcJI4eIfyfL3ANEpiD+uTOFp8oEDa97VWveFIRxF6Db1pmW2IhGLx8TiZXs6i+6MoCTkeV1GQLPA7US84yVPlJpSupU7pMQC3PgeRhAoKnc1wqE2S0OMYIy2lqyDlx5t0nz1G0WIUaXIWKZCq6ntXbNxwijMmhCBM1fwLAmQVPpWN49Sby+UMx1iPQVxC0RsEG0YRV+Sg0nlTU36CL54LKjUexSjAOo7jKC5770VBFB8FkLE8jyDCW4LYLdHoF73zwYduT4gJYFdQHEA2Yj+t2h05uyTeHCwGiRM3oLJCXLqh77kUqhHfN6NoOg3wUcJYFELJOcfx24kUkYoE+G/2qpIffrKIcM+8Gfbm2Idgyrjk6tKVDp8lE6T9ki5lAUxCKT61V9uwvyI4n7IUyPSWuVJTU6kyT4CAH1DcZ7mqTzGDrQUnF1uDi+m6rKYIBLKenBEIa+lyNYsBn9Wve4PB2VfFUTI1VwSrTyVjN50smytwSbXAqQkl3s8oMpxiDN+hQuP/6uPXT/1Rr5KFTG2Dw8PDTZcz655DASkiXckCmFewGey8YPO0u/RijMPLpevh3gqM+iSJNHWAhCuXcqNSZn6W9MNlAg0Gh7hQIhbRSnhVRHdz15XRzPHMfFW1FfgygdAA+ecRJVyN3I2v2r6bOMLJytzQw9FkW3pvR3Zz43TkooK2ugvE0XqJ9eHZl8ve7fDsqlc8RrwAfPv0dyJRlPDqcLwG5+UGyP89elLbIm1fVI7ALON6bhdpt8JJuRLxnTEjyajvOMZA4C8vEnSUGFXxg2gqVWweIYABw3Hl9QzyzHmJ4VYA+zF2lzPwd7GMQtBBdRa7IZ1E8ULxPkENoTh3Oev+BMp0J2r9lAJKRno6ZMZkzC8h95dEW33LuW5/AsLdXC/hmHFlUP1M0g16v41uu4N+918VdT/oh4jNCM0aOLzC8TpLhZlLkRuiQs3m9wZRyNH9jHgzBKwumgbRGO5c/WNdyBxxgRkdjBMK9wwWIZfSyCMuw0oBMHtxRCnyyWQCvodpJwlKfTQ4O6ZS4NconrsxRwlqGxAg+xiJ3lEYCN0Yx8jDS4YgA1NjxryvQcSn79CMsSV9V6tNCZslY77zahKlj5c1238rNNYIpQn8WI26Kkapz0N3CXvjx1MH+lN0lJDAB48XMAm6UhLKnVMu6dmJQnwfhz1Fcoi+8yiZpmUYZlUOoak25BBaeruRzTaaYtg0baMtJ412o26l66bZbGWk9ZyrmcqynJajZh2j2cg0WFZO21YEtlXPTXByE/Jh3X7Gxs3JVbTr+VDS2m0rFQuj1BvO7yi1ttFUIAAeOVM7N8DIJptqlNvXbmXibSMX5GQ8XOQf5fZsmnUaG82OPHnR0o0p7uerO/uWKgBQedZQ7WxkRAe97aiho0iQwt7IN9v3dPhHJ6eHznrKZnndU4y6XBCUUECRxvtqAnRGB34OUr4O+vVXsnH8TDb7QyXyO/kjLS15jZUmlvoxL4DirYmlapmyUmquCu1mELn+9mZzVyva2dZUboli5zmyfwkpgV7woeC+LibOJmWh/ACWlO/RWzPDMGOiS+g3sCSBXbHD992ev9Y5HtdXuPZC104rz6znJUkFe4vEAjh5i595vkzorGijUl+uehmqqdDKVhx1Cidt4R64BUI4TvibXVee9+r5T0vPf8gxNwjGrjffmmYDlT6qXSgv7EyfDbIX00iJhgiqUBN1SCvjyjHd5WOhZ6TibWGHd6lG/tQJRKoVKm3xEMcpYrRgYmk+2zcb26ZEVNlwO6tHJapiNXp9PUK7YluSDWWpAEOx+Sk3i6/eJ6UQqLcO/juAZOWPD6JtGhAKfHDPKL/2qCeFnQErZLw4KMpPKvz3i6iHf1tRqawWVK3cGMlbUeGJVHzzS5IY8OvRB8ZiMk4Y1vaZOxbJul9F+8Z+RbQ75wls8chLKIpC2WcJzkxCJO9bPHVTszUV15RCcGvCsPJh7LlLbnJXPMJeYsagAaaatLS8zQVF0Qv+3dkk6MrHqT//F0+8//iRMqpLn/4AnfXz2fXW2cenn3yHzUTM5EpJKqcqTsTA8ZS+v/6ZoaDuBzkWWX/iQwTEDfbZVtuf47Uf3Yf7xaez7J7KawLWgUQ8sux7+3nGvhg7mZvqdWyvundA2TrA71W2/5PAxS9m/O1W0/Uaw4tlAFcCWhsn3py/N9LscnNQK7Ie+GQFDf3hlv8ekBfA3QJWJkkQXJJHfLP3/qAG9M9YebaAtWdwzQncNSebme8HclLX9YMafG7nZFEElSPekPwf) won't work in `main` (the blue circles (valves?) won't show up, but on this branch they appear.

I eventually tracked it down to a subtle bug in `I3dmLoader`, where `POSITION_QUANTIZED` was being parsed. Instead of creating a new `Float32Array` for the result, the dequantized values were being overwritten in place to a `Uint32Array`, and other typos turned the computed value into a `NaN` (and apparently writing `uint16Array[i] = NaN` stores a 0 :open_mouth:).

Furthermore, the one spec for this case checks the type of the result but not the position values. I added a check to the spec to check for the model matrix (which is derived from the quantized positions)

Here's what the above Sandcastle is supposed to look like (the tile in question was a `.cmpt` with a single `.b3dm` and several `.i3dm` files inside)
image

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ptrgags!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/fix-i3dm-quantization/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Comment by ptrgags** [comment]
@j9liu could you review this when you get a chance?

**Comment by j9liu** [comment]
@ptrgags I'll review soon, but looks like some unit tests are failing that may be related to #10593. Can you check that you're on the most recent commit in `main`?

**Comment by ptrgags** [comment]
@j9liu just synced with main, hopefully that fixes things in Travis!

File: Specs/Scene/ModelExperimental/I3dmLoaderSpec.js [hunk 1]
```diff
@@ -267,22 +268,13 @@ describe("Scene/ModelExperimental/I3dmLoader", function () {
...
-      expect(transform[Matrix4.COLUMN0ROW2]).toEqual(0.0);
-      expect(transform[Matrix4.COLUMN1ROW2]).toEqual(0.0);
-      expect(transform[Matrix4.COLUMN2ROW2]).toEqual(1.0);
-      expect(transform[Matrix4.COLUMN3ROW2]).toEqual(4081608.4380407534);
-      expect(transform[Matrix4.COLUMN0ROW3]).toEqual(0.0);
-      expect(transform[Matrix4.COLUMN1ROW3]).toEqual(0.0);
-      expect(transform[Matrix4.COLUMN2ROW3]).toEqual(0.0);
-      expect(transform[Matrix4.COLUMN3ROW3]).toEqual(1.0);
+      // prettier-ignore
+      expect(transform).toEqualEpsilon(new Matrix4(
```
Comment (j9liu) [inline Specs/Scene/ModelExperimental/I3dmLoaderSpec.js:272]: It seems like the epsilon is missing from this function -- maybe add one or change this to `toEqual`

**Comment by j9liu** [comment]
@ptrgags I can confirm that the Sandcastle works for me, and the tests pass as well. Just one comment about a unit test

**Comment by ptrgags** [comment]
@j9liu updated!

**Comment by j9liu** [comment]
@ptrgags looks good, thanks!

**Pull request merged**